### ISSUE TITLE
Input image size

### ISSUE THREAD

**Issue by cocacola0** [sconlyshootery/FeatDepth#40]
I want to train using a custom dataset of size 1600x900. Due to low available computing power, I have to lower the image size. So far, I have tried with the following: 320x1024 (kitti size, does not keep aspect ratio), 576x1024(keeps aspect ratio). The problem is when I try with certain sizes (ex: 288x512), I get black results. Other sizes (ex: 360x640) produces compilation error. 
I did the scaling of intrinsic parameter:
```
K[0, :] *= self.current_width/self.original_width
K[1, :] *= self.current_height/self.original_height
```
And I modified in net.py, line 148:
`pose_feats = {f_i: F.interpolate(inputs["color_aug", f_i, 0], [self.opt.height, self.opt.width], mode="bilinear", align_corners=False) for f_i in self.opt.frame_ids}`

My question is: Is there any thinking that goes behind choosing the right size? I have observed that the width and height are multiples of 64 in every case that works, can you confirm if this logic makes sense ?

**Comment by sconlyshootery** [comment]
https://github.com/sconlyshootery/FeatDepth/blob/db7344ae9d9cefafc92983fc8c5b30920751be68/mono/datasets/kitti_dataset.py#L19
Did you modify the normalized intrinsics here

**Comment by sconlyshootery** [comment]
mono/datasets/folder_dataset.py
BTW, you can refer to this code to build your own dataloader for your custom dataset.

**Comment by cocacola0** [comment]
Thank you for your kind response! My custom dataset is "Nuscenes" and there is camera intrinsic parameter for every sample, so I can call a function get_intrininsics in the __getitem__ of the dataloader. As you said, I have build a custom dataloader for Nuscenes, very similar to your cityscape_dataset.py. However, I don't think that the problems lies in the dataloader, since this would not explain why I can get good results for some input sizes.

**Comment by rxqy** [comment]
Hi @cocacola0 . 
I also met a similar problem here. The input resolution for posecnn is hard coded here.
https://github.com/sconlyshootery/FeatDepth/blob/db7344ae9d9cefafc92983fc8c5b30920751be68/mono/model/mono_fm_joint/net.py#L148

I think that's why you can get reasonable results when you fix the input aspect ratio, but black with other sizes.

**Comment by cocacola0** [comment]
Thank you @rxqy . I have already removed the hard coded resolution from there.

**Comment by sconlyshootery** [comment]
Someone tell me that the issue can be solved in this way, 'Later I figured it out that there was nothing wrong with the code. Due to a change in nvidia-driver version, I was facing this issue. I was able to solve this issue by installing numba==0.43.1. ’
You may try.

**Comment by sconlyshootery** [comment]
I will close this issue, you can reopen it if you have further question.
